pkg/loop/lggr_broker.go

```diff
@@ -1,42 +1,96 @@
 package loop
 
 import (
+	"context"
 	"fmt"
 	"io"
+	"net"
+	"sync/atomic"
 
 	"github.com/hashicorp/go-plugin"
 	"google.golang.org/grpc"
 
 	"github.com/smartcontractkit/chainlink-relay/pkg/logger"
+	"github.com/smartcontractkit/chainlink-relay/pkg/utils"
 )
 
+type broker interface {
+	Accept(id uint32) (net.Listener, error)
+	Dial(id uint32) (conn *grpc.ClientConn, err error)
+	NextId() uint32
+}
+
+var _ broker = (*atomicBroker)(nil)
+
+// An atomicBroker implements [broker] and is backed by a swappable [*plugin.GRPCBroker]
+type atomicBroker struct {
+	broker atomic.Pointer[plugin.GRPCBroker]
+}
+
+func (a *atomicBroker) store(b *plugin.GRPCBroker) { a.broker.Store(b) }
+
+func (a *atomicBroker) Accept(id uint32) (net.Listener, error) {
+	return a.broker.Load().Accept(id)
+}
+
+func (a *atomicBroker) Dial(id uint32) (conn *grpc.ClientConn, err error) {
+	return a.broker.Load().Dial(id)
+}
+
+func (a *atomicBroker) NextId() uint32 {
+	return a.broker.Load().NextId()
+}
+
 type lggrBroker struct {
```

Comment:
aside: why is the log broker such an important concept throughout the code? it's used all over the place but i don't understand why a broker is needed in the first place.

Reply:
The [`plugin.GRPCBroker`](https://pkg.go.dev/github.com/hashicorp/go-plugin#GRPCBroker) is fundamentally how we connect our grpc clients/servers, both initially and to multiplex as we wire things back and forth.

Reply:
keep helping me - why do we need the logs to be going over grpc? why does the node need to care about the logs from a LOOP implementation?

Reply:
that's good. but i'm still confused.

why is the node providing a logger? is it solely for signature compatibility?

and how is that even possible in terms of dependencies? the logger in the node is a interfaced defined in the `chainlink` repo while the logger used in this code is defined in this repo. 

are we copy and pasting interface definitions across repos and hashicorp magically un/marshals between them?

Reply:
AFAIK it is just piping raw lines through (and prefixing them), that's why we have to configure the logger in the plugins with the same options from the host.

Reply:
The Logger interface in this repo being a subset of core is not directly related to how the data is moving around in this case.

Reply:
i'm still confused a simple level.  what is the utility of the lggrBroker?

is a `lggrBroker` a thing that deals out loggers? (that's my first impression from the name) or a GRPC broker impl that happens to log internally?

some comments on the struct definition would help me a lot

Reply:
It's just a broker with helper methods that need to log. I'll find a better name.,

Reply:
Renamed :shrug: 
```go
// brokerExt extends a broker with various helper methods.
type brokerExt struct {
```